cpp/tests/partitioning/hash_partition_test.cpp

```diff
@@ -361,4 +365,80 @@ TEST_F(HashPartition, FixedPointColumnsToHash)
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(first_result->get_column(1).view(), first_input.column(1));
 }
 
+TEST_F(HashPartition, ListWithNulls)
+{
+  using lcw = cudf::test::lists_column_wrapper<int32_t>;
+
+  lcw to_hash{{{1}, {2, 2}, {0}, {1}, {2, 2}, {0}}, nulls_at({2, 5})};
+  fixed_width_column_wrapper<int32_t> first_col({7, 8, 9, 10, 11, 12});
+  fixed_width_column_wrapper<int32_t> second_col({13, 14, 15, 16, 17, 18});
+  auto const first_input  = cudf::table_view({to_hash, first_col});
+  auto const second_input = cudf::table_view({to_hash, second_col});
+
+  auto const columns_to_hash = std::vector<cudf::size_type>({0});
+
+  cudf::size_type const num_partitions = 3;
+  auto const [first_result, first_offsets] =
+    cudf::hash_partition(first_input, columns_to_hash, num_partitions);
+  auto const [second_result, second_offsets] =
+    cudf::hash_partition(second_input, columns_to_hash, num_partitions);
+
+  // Expect offsets to be equal and num_partitions in length
+  EXPECT_EQ(static_cast<size_t>(num_partitions), first_offsets.size());
+  EXPECT_TRUE(std::equal(
+    first_offsets.begin(), first_offsets.end(), second_offsets.begin(), second_offsets.end()));
+
+  // Expect same result for the hashed columns
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(first_result->get_column(0).view(),
+                                 second_result->get_column(0).view());
+}
+
+TEST_F(HashPartition, StructofStructWithNulls)
+{
+  //  +-----------------+
+  //  |  s1{s2{a,b}, c} |
+  //  +-----------------+
+  // 0 |  { {1, 1}, 5}  |
+  // 1 |  { {1, 1}, 5}  |
+  // 2 |  { {1, 2}, 4}  |
+  // 3 |  Null          |
+  // 4 |  { Null,   4}  |
+  // 5 |  { Null,   4}  |
+
+  auto const to_hash = [&] {
+    auto a  = fixed_width_column_wrapper<int32_t>{1, 1, 1, 0, 0, 0};
+    auto b  = fixed_width_column_wrapper<int32_t>{1, 1, 2, 0, 0, 0};
+    auto s2 = structs_col{{a, b}, nulls_at({4, 5})};
+
+    auto c = fixed_width_column_wrapper<int32_t>{5, 5, 4, 0, 4, 4};
+    std::vector<std::unique_ptr<cudf::column>> s1_children;
+    s1_children.emplace_back(s2.release());
+    s1_children.emplace_back(c.release());
+    auto const null_it = nulls_at({3});
+    return structs_col(std::move(s1_children), std::vector<bool>{null_it, null_it + 6});
```

Comment:
```suggestion
    return structs_col(std::move(s1_children), null_at(3));
```

Reply:
Okay, so this doesn't work actually with this error:
```
/home/nfs/dgala/cudf/cpp/tests/partitioning/hash_partition_test.cpp:417:58: error: no matching function for call to 'cudf::test::structs_column_wrapper::structs_column_wrapper(std::remove_reference<std::vector<std::unique_ptr<cudf::column> >&>::type, thrust::transform_iterator<cudf::test::iterators::nulls_at(Iter, Iter) [with Iter = __gnu_cxx::__normal_iterator<const int*, std::vector<int> >]::<lambda(auto:33)>, thrust::counting_iterator<int>, thrust::use_default, thrust::use_default>)'
  417 |     return structs_col(std::move(s1_children), null_at(3));
```

Reply:
Still the same error:
```
/home/nfs/dgala/cudf/cpp/tests/partitioning/hash_partition_test.cpp:419:49: error: no matching function for call to 'cudf::test::structs_column_wrapper::structs_column_wrapper(<brace-enclosed initializer list>, thrust::transform_iterator<cudf::test::iterators::nulls_at(Iter, Iter) [with Iter = __gnu_cxx::__normal_iterator<const int*, std::vector<int> >]::<lambda(auto:33)>, thrust::counting_iterator<int>, thrust::use_default, thrust::use_default>)'
  419 |     return structs_col({s1_children}, null_at(3));
```